### PR TITLE
Fix pricing breakdown issues

### PR DESCRIPTION
**Pull request by robertsreberski**
Related to https://github.com/Automattic/wp-calypso/pull/86896

## Proposed Changes

Fix the issues that show incorrect first subtotal (before discounts):
image

## Testing Instructions

* Go to /checkout/jetpack/jetpack_ai_yearly and /checkout/jetpack/jetpack_social_advanced_yearly
* Select "Two Year" plan, and make sure the pricing breakdown displays correctly

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/wpcom-checkout/src/transformations.ts [hunk 1]
```diff
@@ -280,12 +287,48 @@ function getYearlyVariantFromProduct( product: ResponseCartProduct ) {
...
 		return total + product.item_original_subtotal_integer;
 	}, 0 );
 }
 
+export function getSubtotalWithDiscounts( responseCart: ResponseCart ): number {
+	return responseCart.products.reduce( ( total, product ) => {
+		const isJetpack = isJetpackProduct( product ) || isJetpackPlan( product );
+
+		if ( isJetpack && hasProductFreeMonthTrial( product ) ) {
+			return total + product.item_original_subtotal_integer;
```
Comment (sirbrillig) [inline packages/wpcom-checkout/src/transformations.ts:325]: Can you explain why this change exists? If the subtotal is wrong, it seems like something we should fix on the backend, but I'm sure there's something I'm just misunderstanding.

Comment (robertsreberski) [reply packages/wpcom-checkout/src/transformations.ts:325]: In the backend, it's correct that the first introductory term will be $0. However, from the user perspective we want to achieve the following:
image

Comment (CodeyGuyDylan) [reply packages/wpcom-checkout/src/transformations.ts:325]: I wonder if it would be worth it to add a new value to the backend to handle situations such as this 🤔

Comment (sirbrillig) [reply packages/wpcom-checkout/src/transformations.ts:325]: Oh, I see, so the idea is to have this function return "the subtotal without discounts including multi-year discounts" even though multi-year discounts are not tracked the way we track normal discounts. I didn't mean this was a bad approach, but maybe we should add a comment block to this method or use slightly more explicit naming to explain the situation. Although it looks like maybe https://github.com/Automattic/wp-calypso/pull/86914 handled this in a different way.
